Makefile

```diff
@@ -105,23 +110,39 @@ database-validate-mapping: database-drop-schema ### Validate mapping in Zenstruc
 
 .PHONY: database-drop-schema
 database-drop-schema: vendor ### Drop database schema
+	@$(MAKE) --no-print-directory docker-start-if-not-running
 	@${DOCKER_PHP} bin/doctrine orm:schema-tool:drop --force
 	@${DOCKER_PHP} vendor/bin/doctrine-migrations migrations:sync-metadata-storage # prevents the next command to fail if migrations table does not exist
 	@${DOCKER_PHP} bin/doctrine dbal:run-sql "TRUNCATE doctrine_migration_versions" --quiet
 
 .PHONY: composer
 composer: ### Run composer command
+	@$(MAKE) --no-print-directory docker-start-if-not-running
 	@${DOCKER_PHP_WITHOUT_XDEBUG} /usr/bin/composer $(ARGS)
 
 vendor: $(DOCKER_PHP_CONTAINER_FLAG) composer.json $(wildcard composer.lock) $(wildcard .env)
+	@$(MAKE) --no-print-directory docker-start-if-not-running
 	@${DC_EXEC} -e SYMFONY_REQUIRE=${SYMFONY_REQUIRE} ${PHP} php -d 'xdebug.mode=off' /usr/bin/composer update ${COMPOSER_UPDATE_OPTIONS}
 	@touch -c $@ composer.json .env composer.lock
 
+.PHONY: docker-start-if-not-running
+docker-start-if-not-running: ### some xxx
+	@if [ -f "$(DOCKER_PHP_CONTAINER_FLAG)" ] ; then \
```

Comment:
this line ensures the containers have been built at lease one time, so that the `--no-build` flag will not create an error.

otherwise, if `DOCKER_PHP_CONTAINER_FLAG` a docker build will be triggered